src/api/Intelligence.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @flow
+ * @file Helper for the Box AI API endpoint
+ * @author Box
+ */
+
+import Base from './Base';
+import { ERROR_CODE_INTELLIGENCE } from '../constants';
+import type { BoxItem } from '../common/types/core';
+import type { ElementsErrorCallback } from '../common/types/api';
+
+class Intelligence extends Base {
+    /**
+     * @property {Function}
+     */
+    successCallback: Function;
+
+    /**
+     * @property {Function}
+     */
+    errorCallback: ElementsErrorCallback;
+
+    /**
+     * API endpoint to ask ai a question
+     *
+     * @param {string} prompt - Question
+     * @param {Array<object>} items - Array of items to ask about
+     * @return {Promise}
+     */
+    async ask(prompt: string, items: Array<BoxItem>, successCallback: Function, errorCallback: ElementsErrorCallback) {
+        this.errorCode = ERROR_CODE_INTELLIGENCE;
+        if (!prompt) {
+            throw new Error('Missing prompt!');
+        }
+
+        if (!items || items.length === 0) {
+            throw new Error('Missing items!');
+        }
+
+        items.forEach(item => {
+            if (!item.id || !item.type) {
+                throw new Error('Invalid item!');
+            }
+        });
+
+        const url = `${this.getBaseApiUrl()}/ai/ask`;
+        const body = {
+            data: {
+                mode: 'single_item_qa',
+                prompt,
+                items,
+            },
+        };
+
+        this.successCallback = successCallback;
+        this.errorCallback = errorCallback;
```

Comment:
Why do these need to be set as variables of the class?

Reply:
Should we also validate that successCallback and errorCallback are valid or have a default value?

Reply:
pattern is used in other clients, its been set for the base class to use

Reply:
https://github.com/box/box-ui-elements/blob/master/src/api/Base.js#L182
https://github.com/box/box-ui-elements/blob/master/src/api/Base.js#L194
Base class is handling the validation of those callbacks